src/quantuminspire/qiskit/backend_qx.py

```diff
@@ -171,25 +177,31 @@ def _submit_experiment(self, experiment: QasmQobjExperiment, number_of_shots: in
         measurements = self._collect_measurements(experiment)
         user_data = {'name': experiment.header.name, 'memory_slots': experiment.header.memory_slots,
                      'creg_sizes': experiment.header.creg_sizes, 'measurements': measurements}
-        job_id = self.__api.execute_qasm_async(compiled_qasm, backend_type=self.__backend,
-                                               number_of_shots=number_of_shots, project=project,
-                                               job_name=experiment.header.name, user_data=json.dumps(user_data),
-                                               full_state_projection=full_state_projection)
-        return job_id
-
-    def get_experiment_results(self, qi_job: QIJob) -> List[ExperimentResult]:
+        quantum_inspire_job = self.__api.execute_qasm_async(compiled_qasm, backend_type=self.__backend,
+                                                            number_of_shots=number_of_shots, project=project,
+                                                            job_name=experiment.header.name,
+                                                            user_data=json.dumps(user_data),
+                                                            full_state_projection=full_state_projection)
+        return quantum_inspire_job
+
+    def get_experiment_results(self, qi_job: QIJob, only_latest_run: bool = True) -> List[ExperimentResult]:
```

Comment:
As you can see from the comments on the test scripts, the meaning of the second argument is not clear. Actually, you are trying to merge two different functions into one here. It might be preferable to make a new function `get_experiment_results_from_all_runs()` to make the meaning explicit.

Reply:
Ok, I'll this one in get_experiment_results_from_latest_run and get_experiment_results_from_all_jobs